parser/strconv.go

```diff
@@ -16,6 +17,32 @@ func ParseTime(lit string) (*ast.DateTimeLiteral, error) {
 	}, nil
 }
 
+// ParseSignedDuration will convert a string into a possibly negative DurationLiteral.
+func ParseSignedDuration(lit string) (*ast.DurationLiteral, error) {
+	if len(lit) > 0 && lit[0] == '-' {
```

Comment:
We should use https://golang.org/pkg/unicode/utf8/#DecodeRuneInString here to ensure that we are utf-8 safe.

Reply:
lit[0] == '-' is utf8 safe because it is at the start of the string, but I can see how using DecodeRuneInString is more standard, I'll switch over.